app/src/main/java/app/revanced/integrations/patches/BackToExitPatch.java

```diff
@@ -0,0 +1,42 @@
+package app.revanced.integrations.patches;
+
+import com.google.android.apps.youtube.app.watchwhile.WatchWhileActivity;
+import app.revanced.integrations.utils.LogHelper;
+
+public class BackToExitPatch {
+    /**
+     * State whether the scroll position reaches the top.
+     */
+    public static boolean isHomeScrolledTop = false;
+
+    /**
+     * Handle the event of clicking the back button.
+     *
+     * @param activity The activity, the app is launched with to finish.
+     */
+    public static void exitOnBackPressed(WatchWhileActivity activity) {
+        if (!isHomeScrolledTop) return;
+
+        activity.finish();
+        
+        LogHelper.printDebug(() -> "Closed activity");
+    }
+
+    /**
+     * Handle the event, when it is being scrolled.
+     */
+    public static void onStartScrollView() {
+        isHomeScrolledTop = false;
+
+        LogHelper.printDebug(() -> "Started scrolling");
+    }
+
+    /**
+     * Handle the event of stopping to scroll.
+     */
+    public static void onStopScrollView() {
```

Comment:
From what the code and comments convey, is that when scrolling stops, the next click on the back button will close the app. This should not be the case. Instead it should only close, when - as the variable name `isHomeScrolledTop` suggests - the feed is scrolled to top.

Reply:
So what you suggest? To change the variables/methods names?

Reply:
Anyway...the current code close the activity immediately after the list is entirely scrolled to the up (until the first video).

So should be good.

Reply:
> So what you suggest? To change the variables/methods names?

If the behavior works as intended, the comment and method name should be adjusted to reflect what they really do.